src/eduid_userdb/element.py

```diff
@@ -416,20 +416,26 @@ class PrimaryElementList(ElementList):
 
     Provide methods to add, update and remove elements from the list while
     maintaining some governing principles, such as ensuring there is exactly
-    one primary element in the list (except if the list is empty).
+    one primary element in the list (except if the list is empty or there are
+    no confirmed elements).
 
     :param elements: List of elements
-    :param require_verified_primary: Is the primary element required to be verified?
     :type elements: [dict | Element]
-    :type require_verified_primary: bool
     """
-    def __init__(self, elements, require_verified_primary=True):
-        self.require_verified_primary = require_verified_primary
-        try:
-            if elements:
-                assert self._get_primary(elements) is not None
-        except PrimaryElementViolation:
-            raise
+    def __init__(self, elements):
+        verified = [e for e in elements if e.is_verified]
+        if len(verified) > 0:
+            try:
```

Comment:
Actually again we hit a maximum recursion, _check_primary calls self.verified that calls PrimaryElementList.__init__